src/types.ts

```diff
@@ -1,9 +1,14 @@
 import {CompilerOptions} from 'typescript';
 
+export interface IdMetadata {
+  descriptor: string;
+  key: string;
+}
+
 export interface CodeSample {
   language: 'js' | 'ts' | null;
-  id: string;
-  sectionId: string | null;
+  id: IdMetadata;
+  sectionId: IdMetadata | null;
```

Comment:
Threading [your comment here @danvk](https://github.com/danvk/literate-ts/pull/37#pullrequestreview-947139357): 

> My one bit of high level feedback is whether it would be possible to move the "descriptor" into the `CodeSample` interface, rather than making `Id` an object. The location in the source file feels like a property of the sample, rather than a property of the identifier. NBD if it's a pain, this is clearly a win for usability!

I do like your suggestion and would prefer it... but I'm not seeing clearly how to wrangle it into the types, since there are _two_ cases of using it in the `CodeSample`. Could you help me please? 🙂

Reply:
Well shame on me for not including comments here, but what is `sectionId` used for? Is it used for anything?

Reply:
Ha there we go! It's never set. Nice.

Reply:
Nice, doesn't seem like it is!